### PR TITLE
CI: Release automation fixes

### PR DESCRIPTION
**Pull request by dsotirakis**
**What this PR does / why we need it**:

After testing new release automations, some small tweaks were needed in order for them to work properly.
Those are:

* Download files from the correct bucket upon publishing.
* Make publish depend on artifacts publish.
* Make sure all `*.msi` files are stored to the correct path.

### CODE REVIEW

**Comment by dsotirakis** [comment]
This is actually a backport of https://github.com/grafana/grafana/pull/44194

File: .drone.yml [hunk 1]
```diff
@@ -3341,8 +3344,8 @@ steps:
   - .\grabpl.exe windows-installer --edition oss --packages-bucket grafana-downloads-test
     v7.3.0-test
   - $$fname = ((Get-Childitem grafana*.msi -name) -split "`n")[0]
-  - gsutil cp $$fname gs://grafana-downloads-test/oss/release/
-  - gsutil cp "$$fname.sha256" gs://grafana-downloads-test/oss/release/
+  - gsutil cp $$fname gs://grafana-downloads-test/v7.3.0-test/oss/release/
```
Comment (malcolmholmes) [inline .drone.yml:3347]: woah, what's going on here?

File: .drone.yml [hunk 2]
```diff
@@ -2797,7 +2798,8 @@ steps:
   name: grabpl
 - commands:
   - printenv GCP_KEY | base64 -d > /tmp/gcpkey.json
-  - ./bin/grabpl store-packages --edition oss --gcp-key /tmp/gcpkey.json ${DRONE_TAG}
```
Comment (malcolmholmes) [inline .drone.yml:2800]: good to remove this unnecessary passing in of gcp-key. It just complicates the config

**Review by malcolmholmes — APPROVED** [review]

File: .drone.yml (hunk 1 above)
Comment (dsotirakis) [reply .drone.yml:3347]: This is for test releases, time to remove them completely (in another PR)

**Pull request merged**